File: src/main/java/exam/bfsdfs/BaekJoon1260.java
package exam.bfsdfs;

import org.omg.PortableInterceptor.INACTIVE;

import java.util.LinkedList;
import java.util.Queue;
import java.util.Scanner;
import java.util.Stack;

public class BaekJoon1260 {

    static boolean visited[];
    static int[][] maps;
    static int N,M,V;

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        N = scanner.nextInt();
        M = scanner.nextInt();
        V = scanner.nextInt();

        visited = new boolean[N+1];
        maps = new int[N+1][N+1];

        for(int i=1; i<=M; i++){
            int start = scanner.nextInt();
            int arrived = scanner.nextInt();
            maps[start][arrived] = 1;
            maps[arrived][start] = 1;
        }
        System.out.println("DFS Start");
        dfs(V);
        visited = new boolean[N+1];
        System.out.println("BFS Start");
        bfs(V);

    }

    public static void dfs(int point){
        Stack stack = new Stack();
        stack.push(point);

        while (!stack.isEmpty()){
            int index = (int) stack.pop();
            if(visited[index]) continue;
            visited[index] = true;
            System.out.print(index + "->");
            for(int i=N; i>=1; i--){
                if(maps[index][i] == 1 && visited[i] == false){
                   stack.push(i);
                }
            }
        }

    }
    public static void bfs(int point){
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(point);
        while (!queue.isEmpty()){
            int index = queue.poll();
            visited[index] = true;
            System.out.print(index + "->");
            for(int i=1; i<=N; i++){
                if(maps[index][i] == 1 && visited[i] == false){
                    queue.offer(i);
                    visited[i] = true;
                }
            }
        }
    }

}

File: src/main/java/exam/bfsdfs/BaekJoon1260.kt
package exam.bfsdfs
import java.util.*

object BaekJoon1260 {
    lateinit var visited: BooleanArray
    lateinit var maps: Array<IntArray>
    var N = 0
    var M = 0
    var V = 0
    @JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        N = scanner.nextInt()
        M = scanner.nextInt()
        V = scanner.nextInt()
        visited = BooleanArray(N + 1)
        maps = Array(N + 1) { IntArray(N + 1) }
        for (i in 1..M) {
            val start = scanner.nextInt()
            val arrived = scanner.nextInt()
            maps[start][arrived] = 1
            maps[arrived][start] = 1
        }

        dfs(V)
        println()
        visited = BooleanArray(N + 1)
        bfs(V)
    }

    private fun dfs(point: Int) {
        val stack: Stack<Int> = Stack<Int>()
        stack.push(point)
        while (!stack.isEmpty()) {
            val index = stack.pop() as Int
            if (visited[index]) continue
            visited[index] = true
            print("$index ")
            for (i in N downTo 1) {
                if (maps[index][i] == 1 && !visited[i]) {
                    stack.push(i)
                }
            }
        }
    }

    private fun bfs(point: Int) {
        val queue: Queue<Int> = LinkedList()
        queue.offer(point)
        while (!queue.isEmpty()) {
            val index = queue.poll()
            visited[index] = true
            print("$index ")
            for (i in 1..N) {
                if (maps[index][i] == 1 && !visited[i]) {
                    queue.offer(i)
                    visited[i] = true
                }
            }
        }
    }
}